src/GitHub.App/Controllers/UIController.cs

```diff
@@ -142,6 +136,25 @@ public IObservable<UserControl> SelectFlow(UIControllerFlow choice)
             return transition;
         }
 
+        /// <summary>
+        /// Allows listening to the completion state of the ui flow - whether
+        /// it was completed because it was cancelled or whether it succeeded.
+        /// </summary>
+        /// <returns>true for success, false for cancel</returns>
+        public IObservable<bool> ListenToCompletionState()
+        {
+            completion = new Subject<bool>();
+            return completion;
+        }
+
+        void End(bool success)
+        {
+            uiProvider.RemoveService(typeof(IConnection));
+            transition.OnCompleted();
+            completion.OnNext(success);
```

Comment:
It's an optional listener, so being null is fine, I'll just add the ? there that I forgot.